draft-pda-protocol.md

```diff
@@ -711,17 +717,27 @@ struct {
 } PDAAggregateSubReq;
 ~~~
 
-The `helper_share` field is the helper's encrypted input share as it appeared in
-the report uploaded by the client. [OPEN ISSUE: We usually only need to send
-this in the first aggregate request. Shall we exclude it in subsequent requests
-somehow?] The `time` field should match the timestamp of the corresponding
-report. The remainder of the structure is dedicated to the protocol-specific
-helper share and request parameters used for the current round.
+The `time` and `jitter` fields have the same value as those in the report
+uploaded by the client. Similarly, the `helper_share` field is the helper's
+encrypted input share as it appeared in the report. [OPEN ISSUE: We usually only
+need to send this in the first aggregate request. Shall we exclude it in
+subsequent requests somehow?] The remainder of the structure is dedicated to the
+protocol-specific request parameters.
+
+In order to provide replay protection, the leader is required to send aggregate
+sub-requests in ascending order, where the ordering on sub-requests is
```

Comment:
If the leader is responsible for reordering reports, then it should allow a grace period after an aggregation window starts for late reports to arrive from clients. Unlike Prio v2, the two aggregators don't need to agree on the grace period, since the leader drives aggregations (at the request of a collector, but since those are going to have to be asynchronous anyway I think it's OK for a leader to acknowledge a collect request but not service it until the leader is satisfied the aggregation window's grace period has elapsed). So I think we don't need to put grace period into PDAParams or specify much about what leaders should do, but perhaps we should have a SHOULD statement urging leader implementations to implement a grace period proportional to the aggregation window or other deployment specific details (e.g. special knowledge of the schedule on which clients upload reports)?

Reply:
Good idea, done.